TLM/TLM/Manager/Impl/ParkingRestrictionsManager.cs

```diff
@@ -25,15 +25,29 @@ public class ParkingRestrictionsManager
 
         public bool MayHaveParkingRestriction(ushort segmentId) {
             ref NetSegment segment = ref segmentId.ToSegment();
-            if ((segment.m_flags & NetSegment.Flags.Created) == NetSegment.Flags.None) {
-                return true;
+            if (!segment.IsValid()) {
+                return false;
             }
 
             ItemClass connectionClass = segment.Info.GetConnectionClass();
-
             return connectionClass.m_service == ItemClass.Service.Road && segment.Info.m_hasParkingSpaces;
         }
 
+        public bool MayHaveParkingRestriction(ushort segmentId, NetInfo.Direction finalDir) {
+            ref NetSegment segment = ref segmentId.ToSegment();
+            bool right = (finalDir == NetInfo.Direction.Forward) != segment.m_flags.IsFlagSet(NetSegment.Flags.Invert);
```

Comment:
What about LHT _left-hand traffic_(RHD - _right hand drive_)?

Reply:
I use final dir for that reason
I need to test that though. I only tested invert

Reply:
thanks for reminding me. it does not work 
![image](https://user-images.githubusercontent.com/26344691/205866287-b0830fe3-d1ba-4860-bdee-fa82075eb10d.png)